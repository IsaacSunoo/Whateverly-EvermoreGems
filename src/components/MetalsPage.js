import React from 'react';
import MetalCard from './MetalCard';

function MetalsPage(props)  {
  const metalsCards = props.metals.map(metal => {
    return <MetalCard 
            selectMetal={props.selectMetal} 
            metalId={metal.id} 
            key={metal.id} 
            metalName={metal.name} 
            metalColors={metal.color} 
            metalMakeup={metal.makeup} />
  });
  return (
    <div className='metal-main'>
      <h1>Select a Metal to compliment your gem!</h1>
      <div className='metal-container'>
        {metalsCards}
      </div>
    </div>
  )
}


export default MetalsPage;