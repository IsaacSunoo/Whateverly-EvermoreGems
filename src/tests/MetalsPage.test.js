import React from 'react';
import MetalsPage from '../components/MetalsPage.js';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const metalsCards = jest.fn();


const mockMetal =[{
        "name": "Tungsten",
        "id": 8,
        "color": ["White"],
        "makeup": ["Tungsten"],
        "durable": "True",
        "resizeable": "False"
    }]


describe('MetalsPage', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MetalsPage
        selectMetal={mockMetal} 
        metalId={mockMetal.id} 
        key={mockMetal.id} 
        metalName={mockMetal.name} 
        metalColors={mockMetal.color} 
        metalMakeup={mockMetal.makeup}
      />
    )
  })
  
  it('should match the snapshot with all data passed in' , () =>{
    expect(wrapper).toMatchSnapshot();
  })
});