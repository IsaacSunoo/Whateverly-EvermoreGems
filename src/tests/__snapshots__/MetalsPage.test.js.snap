// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetalsPage should match the snapshot with all data passed in 1`] = `
<div
  className="metal-main"
>
  <Header />
  <h1>
    Select a Metal to compliment your gem!
  </h1>
  <div
    className="metal-container"
  >
    <MetalCard
      key="8"
      metalColors={
        Array [
          "White",
        ]
      }
      metalId={8}
      metalMakeup={
        Array [
          "Tungsten",
        ]
      }
      metalName="Tungsten"
      selectMetal={[MockFunction]}
    />
  </div>
</div>
`;
